ext/pbr_ext/ss.h

```diff
@@ -0,0 +1,18 @@
+#ifndef ss_h__
+#define ss_h__
+
+#include <cstdint>
+
+typedef struct StringStream {
+  uint8_t *buf;
+  int pos;
+  int len;
+} ss_t;
+
+ss_t ss_make(char *s, int len);
+ss_t ss_substream(ss_t& other, int len);
+uint8_t ss_read_byte(ss_t& ss);
+char *ss_read_chars(ss_t& ss, int32_t len);
+#define ss_more(ss)  (ss).pos < (ss).len
```

Comment:
Just because the code is predominantly C-style (structs and functions that operate on them), using C++ only for references and `vector<>`. Is there an advantage to making these member functions? I agree in most code it would be more "normal" to make StringStream a class with member functions, but I'm not so sure here.